src/views/DocumentationTopic.vue

```diff
@@ -9,28 +9,42 @@
 -->
 
 <template>
-  <div :class="{ 'modal-open': quickNavigationStore.state.showQuickNavigation }">
-    <div v-show="quickNavigationStore.state.showQuickNavigation">
-      <QuickNavigationModal />
-    </div>
-    <CodeTheme class="doc-topic-view">
-      <template v-if="topicData">
-        <component
-          :is="enableNavigator ? 'AdjustableSidebarWidth' : 'StaticContentWidth'"
-          v-bind="sidebarProps"
-          v-on="sidebarListeners"
-        >
-          <template #aside="{ scrollLockID, breakpoint }">
-            <div class="doc-topic-aside">
-              <NavigatorDataProvider
-                :interface-language="topicProps.interfaceLanguage"
-                :technology="technology"
-                :api-changes-version="store.state.selectedAPIChangesVersion"
-              >
-                <template #default="slotProps">
+  <CodeTheme class="doc-topic-view">
+    <template v-if="topicData">
+      <NavigatorDataProvider
```

Comment:
I could be wrong, but having the `NavigatorDataProvider` in two places, would initiate its logic twice, fetching the dataset, storing that huge piece of data twice, all the calculation and so on with it.

Reply:
Mmm, that could be true.  If we are going to share the same data and we don't want to re-use the component because performance issues, what is the point of having a NavigatorDataProvider component?

We could move all that logic to the DocumentationTopic component and send that data to the child `QuickNavigationModal` and `Navigator` components

Reply:
We could just add `NavigatorDataProvider` on the root of `DocumentationTopic` so the data is available for `QuickNavigationModal` and `Navigator` components. Does it sound right, @sofiaromorales ?

Reply:
What Dobri mentioned is correct, I made the refactor here f905ee92e646c01349cd2f59abd67761204e68f2. Two tests were failing after moving things around, and I can't tell for sure why, but I resolved it by removing `await flushPromises()` from both. 

Thanks, Marina, for proposing the solution :)

Reply:
One drawback is that now the `NavigatorDataProvider` is always rendered and tries to fetch the navigator data, but DocCRender may be served older `RenderJSON`, which may not have the navigator data.

I think we may need to introduce some conditionals based on `enableNavigator`.

Reply:
@marinaaisa @dobromir-hristov I see how this can be a drawback, but this would end up in an error caught by the NavigatorDataProvider, so the app functionality is not affected. I could open a new issue and work on this in a different pr, or do you think this is a blocker for merging this pr?

Reply:
But it will error in the user's browser. We can prevent that. 

Also your entire functionality depends on this data being there. Not sure what would happen if its missing.

Reply:
I did a refactor to prevent rendering `NavigatorDataProvider` when there's no available index data using the existing logic based on the same piece of logic that conditionally renders the `Navigator` +  I added logic for enabling opening quick navigation depending on `enableNavigator` too. Thx for the suggestion Dobri! [4038fea](https://github.com/apple/swift-docc-render/pull/396/commits/4038fead95b38d39762d9113dd7bc91279ded66a)